protected/controllers/AdminDatasetController.php

```diff
@@ -323,13 +323,21 @@ public function actionUpdate($id)
                 if( ! $model->image->write(Yii::$app->cloudStore, $model->getUuid(), $datasetImage) ) {
                     Yii::log("Error writing file to storage for dataset ".$model->identifier, "error");
                 }
-                // save image
-                if( $model->image->save() ) {
-                    $model->image_id = $model->image->id;
-                }
-                else {
-                    Yii::log(print_r($model->image->getErrors(), true), 'error');
+            }
+
+            // save image
+            if ($model->image) {
```

Comment:
Some behaviours are not taking into account following the deletion made on image. 
So this PR needs to fix/anticipate other issues linked to this action of saving the model